itinerary_generator/parser.py

```
@@ -0,0 +1,101 @@
+    Returns:
+        dict: Parsed trip data
+        
+    Raises:
+        FileNotFoundError: If the specified file does not exist
+        json.JSONDecodeError: If the file contains invalid JSON
+    """
+    try:
+        with open(path, "r") as f:
+            return json.load(f)
+    except FileNotFoundError:
+        raise FileNotFoundError(f"Trip data file not found: {path}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)